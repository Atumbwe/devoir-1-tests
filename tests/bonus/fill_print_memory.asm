nop Fill memory


push 0
next_fill: push 1
    add
    load
    store
    store
    push 99
    sub
    jumpz done_fill
    push 0
    stack
    read
    print
    push 1
    add 
    stack
    load
    store
    push 0
    write
    store
    stack
    write
    stack
    jump next_fill
done_fill: stack



nop Print Memory
    push 99
next_print: jumpz exit
    load
    store
    store
    read 
    print
    pop
    push 1
    sub
    jump next_print
exit: pop
