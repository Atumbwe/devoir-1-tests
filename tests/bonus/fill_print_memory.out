[ 1 -98 0 ]
0
[ 1 -98 1 ]
[ 1 -98 1 ]
[ 1 ]
[ 2 -97 0 ]
1
[ 2 -97 2 ]
[ 2 -97 2 ]
[ 2 ]
[ 3 -96 0 ]
2
[ 3 -96 3 ]
[ 3 -96 3 ]
[ 3 ]
[ 4 -95 0 ]
3
[ 4 -95 4 ]
[ 4 -95 4 ]
[ 4 ]
[ 5 -94 0 ]
4
[ 5 -94 5 ]
[ 5 -94 5 ]
[ 5 ]
[ 6 -93 0 ]
5
[ 6 -93 6 ]
[ 6 -93 6 ]
[ 6 ]
[ 7 -92 0 ]
6
[ 7 -92 7 ]
[ 7 -92 7 ]
[ 7 ]
[ 8 -91 0 ]
7
[ 8 -91 8 ]
[ 8 -91 8 ]
[ 8 ]
[ 9 -90 0 ]
8
[ 9 -90 9 ]
[ 9 -90 9 ]
[ 9 ]
[ 10 -89 0 ]
9
[ 10 -89 10 ]
[ 10 -89 10 ]
[ 10 ]
[ 11 -88 0 ]
10
[ 11 -88 11 ]
[ 11 -88 11 ]
[ 11 ]
[ 12 -87 0 ]
11
[ 12 -87 12 ]
[ 12 -87 12 ]
[ 12 ]
[ 13 -86 0 ]
12
[ 13 -86 13 ]
[ 13 -86 13 ]
[ 13 ]
[ 14 -85 0 ]
13
[ 14 -85 14 ]
[ 14 -85 14 ]
[ 14 ]
[ 15 -84 0 ]
14
[ 15 -84 15 ]
[ 15 -84 15 ]
[ 15 ]
[ 16 -83 0 ]
15
[ 16 -83 16 ]
[ 16 -83 16 ]
[ 16 ]
[ 17 -82 0 ]
16
[ 17 -82 17 ]
[ 17 -82 17 ]
[ 17 ]
[ 18 -81 0 ]
17
[ 18 -81 18 ]
[ 18 -81 18 ]
[ 18 ]
[ 19 -80 0 ]
18
[ 19 -80 19 ]
[ 19 -80 19 ]
[ 19 ]
[ 20 -79 0 ]
19
[ 20 -79 20 ]
[ 20 -79 20 ]
[ 20 ]
[ 21 -78 0 ]
20
[ 21 -78 21 ]
[ 21 -78 21 ]
[ 21 ]
[ 22 -77 0 ]
21
[ 22 -77 22 ]
[ 22 -77 22 ]
[ 22 ]
[ 23 -76 0 ]
22
[ 23 -76 23 ]
[ 23 -76 23 ]
[ 23 ]
[ 24 -75 0 ]
23
[ 24 -75 24 ]
[ 24 -75 24 ]
[ 24 ]
[ 25 -74 0 ]
24
[ 25 -74 25 ]
[ 25 -74 25 ]
[ 25 ]
[ 26 -73 0 ]
25
[ 26 -73 26 ]
[ 26 -73 26 ]
[ 26 ]
[ 27 -72 0 ]
26
[ 27 -72 27 ]
[ 27 -72 27 ]
[ 27 ]
[ 28 -71 0 ]
27
[ 28 -71 28 ]
[ 28 -71 28 ]
[ 28 ]
[ 29 -70 0 ]
28
[ 29 -70 29 ]
[ 29 -70 29 ]
[ 29 ]
[ 30 -69 0 ]
29
[ 30 -69 30 ]
[ 30 -69 30 ]
[ 30 ]
[ 31 -68 0 ]
30
[ 31 -68 31 ]
[ 31 -68 31 ]
[ 31 ]
[ 32 -67 0 ]
31
[ 32 -67 32 ]
[ 32 -67 32 ]
[ 32 ]
[ 33 -66 0 ]
32
[ 33 -66 33 ]
[ 33 -66 33 ]
[ 33 ]
[ 34 -65 0 ]
33
[ 34 -65 34 ]
[ 34 -65 34 ]
[ 34 ]
[ 35 -64 0 ]
34
[ 35 -64 35 ]
[ 35 -64 35 ]
[ 35 ]
[ 36 -63 0 ]
35
[ 36 -63 36 ]
[ 36 -63 36 ]
[ 36 ]
[ 37 -62 0 ]
36
[ 37 -62 37 ]
[ 37 -62 37 ]
[ 37 ]
[ 38 -61 0 ]
37
[ 38 -61 38 ]
[ 38 -61 38 ]
[ 38 ]
[ 39 -60 0 ]
38
[ 39 -60 39 ]
[ 39 -60 39 ]
[ 39 ]
[ 40 -59 0 ]
39
[ 40 -59 40 ]
[ 40 -59 40 ]
[ 40 ]
[ 41 -58 0 ]
40
[ 41 -58 41 ]
[ 41 -58 41 ]
[ 41 ]
[ 42 -57 0 ]
41
[ 42 -57 42 ]
[ 42 -57 42 ]
[ 42 ]
[ 43 -56 0 ]
42
[ 43 -56 43 ]
[ 43 -56 43 ]
[ 43 ]
[ 44 -55 0 ]
43
[ 44 -55 44 ]
[ 44 -55 44 ]
[ 44 ]
[ 45 -54 0 ]
44
[ 45 -54 45 ]
[ 45 -54 45 ]
[ 45 ]
[ 46 -53 0 ]
45
[ 46 -53 46 ]
[ 46 -53 46 ]
[ 46 ]
[ 47 -52 0 ]
46
[ 47 -52 47 ]
[ 47 -52 47 ]
[ 47 ]
[ 48 -51 0 ]
47
[ 48 -51 48 ]
[ 48 -51 48 ]
[ 48 ]
[ 49 -50 0 ]
48
[ 49 -50 49 ]
[ 49 -50 49 ]
[ 49 ]
[ 50 -49 0 ]
49
[ 50 -49 50 ]
[ 50 -49 50 ]
[ 50 ]
[ 51 -48 0 ]
50
[ 51 -48 51 ]
[ 51 -48 51 ]
[ 51 ]
[ 52 -47 0 ]
51
[ 52 -47 52 ]
[ 52 -47 52 ]
[ 52 ]
[ 53 -46 0 ]
52
[ 53 -46 53 ]
[ 53 -46 53 ]
[ 53 ]
[ 54 -45 0 ]
53
[ 54 -45 54 ]
[ 54 -45 54 ]
[ 54 ]
[ 55 -44 0 ]
54
[ 55 -44 55 ]
[ 55 -44 55 ]
[ 55 ]
[ 56 -43 0 ]
55
[ 56 -43 56 ]
[ 56 -43 56 ]
[ 56 ]
[ 57 -42 0 ]
56
[ 57 -42 57 ]
[ 57 -42 57 ]
[ 57 ]
[ 58 -41 0 ]
57
[ 58 -41 58 ]
[ 58 -41 58 ]
[ 58 ]
[ 59 -40 0 ]
58
[ 59 -40 59 ]
[ 59 -40 59 ]
[ 59 ]
[ 60 -39 0 ]
59
[ 60 -39 60 ]
[ 60 -39 60 ]
[ 60 ]
[ 61 -38 0 ]
60
[ 61 -38 61 ]
[ 61 -38 61 ]
[ 61 ]
[ 62 -37 0 ]
61
[ 62 -37 62 ]
[ 62 -37 62 ]
[ 62 ]
[ 63 -36 0 ]
62
[ 63 -36 63 ]
[ 63 -36 63 ]
[ 63 ]
[ 64 -35 0 ]
63
[ 64 -35 64 ]
[ 64 -35 64 ]
[ 64 ]
[ 65 -34 0 ]
64
[ 65 -34 65 ]
[ 65 -34 65 ]
[ 65 ]
[ 66 -33 0 ]
65
[ 66 -33 66 ]
[ 66 -33 66 ]
[ 66 ]
[ 67 -32 0 ]
66
[ 67 -32 67 ]
[ 67 -32 67 ]
[ 67 ]
[ 68 -31 0 ]
67
[ 68 -31 68 ]
[ 68 -31 68 ]
[ 68 ]
[ 69 -30 0 ]
68
[ 69 -30 69 ]
[ 69 -30 69 ]
[ 69 ]
[ 70 -29 0 ]
69
[ 70 -29 70 ]
[ 70 -29 70 ]
[ 70 ]
[ 71 -28 0 ]
70
[ 71 -28 71 ]
[ 71 -28 71 ]
[ 71 ]
[ 72 -27 0 ]
71
[ 72 -27 72 ]
[ 72 -27 72 ]
[ 72 ]
[ 73 -26 0 ]
72
[ 73 -26 73 ]
[ 73 -26 73 ]
[ 73 ]
[ 74 -25 0 ]
73
[ 74 -25 74 ]
[ 74 -25 74 ]
[ 74 ]
[ 75 -24 0 ]
74
[ 75 -24 75 ]
[ 75 -24 75 ]
[ 75 ]
[ 76 -23 0 ]
75
[ 76 -23 76 ]
[ 76 -23 76 ]
[ 76 ]
[ 77 -22 0 ]
76
[ 77 -22 77 ]
[ 77 -22 77 ]
[ 77 ]
[ 78 -21 0 ]
77
[ 78 -21 78 ]
[ 78 -21 78 ]
[ 78 ]
[ 79 -20 0 ]
78
[ 79 -20 79 ]
[ 79 -20 79 ]
[ 79 ]
[ 80 -19 0 ]
79
[ 80 -19 80 ]
[ 80 -19 80 ]
[ 80 ]
[ 81 -18 0 ]
80
[ 81 -18 81 ]
[ 81 -18 81 ]
[ 81 ]
[ 82 -17 0 ]
81
[ 82 -17 82 ]
[ 82 -17 82 ]
[ 82 ]
[ 83 -16 0 ]
82
[ 83 -16 83 ]
[ 83 -16 83 ]
[ 83 ]
[ 84 -15 0 ]
83
[ 84 -15 84 ]
[ 84 -15 84 ]
[ 84 ]
[ 85 -14 0 ]
84
[ 85 -14 85 ]
[ 85 -14 85 ]
[ 85 ]
[ 86 -13 0 ]
85
[ 86 -13 86 ]
[ 86 -13 86 ]
[ 86 ]
[ 87 -12 0 ]
86
[ 87 -12 87 ]
[ 87 -12 87 ]
[ 87 ]
[ 88 -11 0 ]
87
[ 88 -11 88 ]
[ 88 -11 88 ]
[ 88 ]
[ 89 -10 0 ]
88
[ 89 -10 89 ]
[ 89 -10 89 ]
[ 89 ]
[ 90 -9 0 ]
89
[ 90 -9 90 ]
[ 90 -9 90 ]
[ 90 ]
[ 91 -8 0 ]
90
[ 91 -8 91 ]
[ 91 -8 91 ]
[ 91 ]
[ 92 -7 0 ]
91
[ 92 -7 92 ]
[ 92 -7 92 ]
[ 92 ]
[ 93 -6 0 ]
92
[ 93 -6 93 ]
[ 93 -6 93 ]
[ 93 ]
[ 94 -5 0 ]
93
[ 94 -5 94 ]
[ 94 -5 94 ]
[ 94 ]
[ 95 -4 0 ]
94
[ 95 -4 95 ]
[ 95 -4 95 ]
[ 95 ]
[ 96 -3 0 ]
95
[ 96 -3 96 ]
[ 96 -3 96 ]
[ 96 ]
[ 97 -2 0 ]
96
[ 97 -2 97 ]
[ 97 -2 97 ]
[ 97 ]
[ 98 -1 0 ]
97
[ 98 -1 98 ]
[ 98 -1 98 ]
[ 98 ]
[ 99 0 ]
0
-1
-2
-3
-4
-5
-6
-7
-8
-9
-10
-11
-12
-13
-14
-15
-16
-17
-18
-19
-20
-21
-22
-23
-24
-25
-26
-27
-28
-29
-30
-31
-32
-33
-34
-35
-36
-37
-38
-39
-40
-41
-42
-43
-44
-45
-46
-47
-48
-49
-50
-51
-52
-53
-54
-55
-56
-57
-58
-59
-60
-61
-62
-63
-64
-65
-66
-67
-68
-69
-70
-71
-72
-73
-74
-75
-76
-77
-78
-79
-80
-81
-82
-83
-84
-85
-86
-87
-88
-89
-90
-91
-92
-93
-94
-95
-96
-97
-98
